lib/convection/control/stack.rb

```diff
@@ -178,7 +178,16 @@ def apply(&block)
           o[:capabilities] = capabilities
         end
 
-        if exist?
+        ## Execute before create tasks
+        before_task_type = exist? ? :before_update : :before_create
+        @tasks[before_task_type].delete_if do |task|
```

Comment:
I think that we should only call the `before_update` task *if* the stack has changes to be applied